### PR TITLE
Off-route threshold options (master)

### PR DESCRIPTION
**Pull request by Guardiola31337**
## Description

Exposes `Navigator` off-route threshold options in `MapboxNavigationOptions`

Follow up from https://github.com/mapbox/mapbox-navigation-android/pull/2276#issue-344284927

> Have a local branch that adds this changes to `master` (Kotlin). Will cut a separate PR after this gets merged.

- [x] I have added any issue links
- [x] I have added all related labels (`bug`, `feature`, `new API(s)`, `SEMVER`, etc.)
- [x] I have added the appropriate milestone and project boards

### Goal

Make off-route threshold options configurable

### Implementation

Expose `Navigator` off-route threshold options in `MapboxNavigationOptions` and consume it from `MapboxNavigation` when initialized

## Testing

- [x] I have tested changing the `MapboxNavigationOptions` in [`ComponentNavigationActivity`](https://github.com/mapbox/mapbox-navigation-android/blob/master/app/src/main/java/com/mapbox/services/android/navigation/testapp/activity/navigationui/ComponentNavigationActivity.java) from the test app
- [x] I have tested locally (including `SNAPSHOT` upstream dependencies if needed) through testapp/demo app and run all activities to avoid regressions
- [x] I have tested via a test drive, or a simulation/mock location app
- [x] New and existing unit tests pass locally with my changes

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have updated the `CHANGELOG` including this PR

### CODE REVIEW

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigation.java [hunk 1]
```diff
@@ -1022,7 +1023,8 @@ private void initializeForTest(Context context) {
    * to prevent users from removing it.
    */
   private void initialize() {
-    mapboxNavigator = new MapboxNavigator(new Navigator());
+    Navigator navigator = configureNavigator();
```
Comment (RingerJK) [inline libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigation.java]: unnecessary change

Comment (Guardiola31337) [reply libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigation.java]: What do you mean by unnecessary change?

Something like this? 

```suggestion
    mapboxNavigator = new MapboxNavigator(configureNavigator());
```

Comment (Guardiola31337) [reply libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigation.java]: I pushed ☝️ could you confirm that's what you meant when you have a chance @RingerJK?

Comment (RingerJK) [reply libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigation.java]: it's just nit :) now looks better
